### DISCUSSION TITLE
DCM Notes for 2024-09-23

### DISCUSSION THREAD

**Discussion by glopesdev** [bonsai-rx/bonsai#2022] [Dev Club]
## Features

- https://github.com/bonsai-rx/bonsai/pull/1969
- https://github.com/bonsai-rx/bonsai/pull/1956
- https://github.com/bonsai-rx/bonsai/pull/1999
- https://github.com/bonsai-rx/bonsai/pull/1870

## Documentation

- https://github.com/bonsai-rx/docfx-tools/pull/9

**Comment by glopesdev** [comment]
Feedback from DCM:
 - Package manager dialog:
   - If currently installed version is selected, disable the Update / Install button
   - Either improve terminology of "Show dependencies" and / or add tooltip to explain what "dependencies" are or what they are used for
   - Package details link may be broken
   - Consider adding update and version selection buttons to "Installed" tab
   - #2003 
   - Installed tab seems to be using local package metadata so e.g. package deprecation notices are not shown
   - Consider moving to infinite scroll list rather than pages
 - Multi-panel navigation:
   - Try out deletion feature with many panels
   - Keep global undo / redo stack and keep track of "overall" layout (if nothing else works we can just serialize the entire layout state)
   - Make sure to test invalidation of docked panels from nodes inside include workflows which were modified externally

**Reply by PathogenDavid** [reply]
Some extra thoughts I had about the docking stuff that felt too bikesheddy to bring up during dev club:

* Does middle click to open in new tab from the tree view?
* Does middle clicking a tab close it?
* When you were demoing the windows which were opened first as a new window and then docked were using the toolbox window design language which is slightly odd. (Maybe a limitation of the docking library?)
* Pin button feels wrong for workflow windows. If you make the actual toolbox panels into windows they'd probably want to use that since the dock space will grow to fill the whole window.
* Can you tear tabs out into windows? (It seems like yes but I never saw you do it with a window that started as a tab)
